stdlib/jms/src/main/ballerina/jms/Package.md

```diff
@@ -163,3 +163,71 @@ public function main(string... args) {
     }
 }
 ```
+
+### JMS Topic Subscriber
+
+Following is a topic subscriber program that subscribes to a particular JMS topic.
+
+```ballerina
+import ballerina/jms;
+import ballerina/log;
+
+jms:Connection conn = new({
+    initialContextFactory:"bmbInitialContextFactory",
+    providerUrl:"amqp://admin:admin@carbon/carbon?brokerlist='tcp://localhost:5672'"
+});
+
+jms:Session jmsSession = new(conn, {
+    acknowledgementMode:"AUTO_ACKNOWLEDGE"
+});
+
+endpoint jms:TopicSubscriber subscriberEndpoint {
+    session:jmsSession,
+    topicPattern:"BallerinaTopic"
+};
+
+service<jms:Consumer> jmsListener bind subscriberEndpoint {
+    onMessage(endpoint subscriber, jms:Message message) {
+        match (message.getTextMessageContent()) {
+            string messageText => log:printInfo("Message : " + messageText);
+            error e => log:printError("Error occurred while reading message", err = e);
+        }
+    }
+}
+```
+
+### JMS Topic Producer
+
+Following is a topic producer program that publishes to a particular JMS topic
+
+```ballerina
+import ballerina/jms;
+import ballerina/log;
+
+jms:Connection jmsConnection = new({
+    initialContextFactory:"bmbInitialContextFactory",
+    providerUrl:"amqp://admin:admin@carbon/carbon?brokerlist='tcp://localhost:5672'"
+});
+
+jms:Session jmsSession = new(jmsConnection, {
+    acknowledgementMode:"AUTO_ACKNOWLEDGE"
+});
+
+endpoint jms:TopicPublisher topicPublisher {
+    session:jmsSession,
+    topicPattern:"BallerinaTopic"
+};
+
+public function main(string... args) {
+    match (jmsSession.createTextMessage("Hello from Ballerina")) {
+        error e => {
+            log:printError("Error occurred while creating message", err=e);
```

Comment:
We have different spacing style in line 224 and 228. Shall we fix?

Reply:
I think that is because 224 is in a match statement and 228 is in match expression. We can remove the braces which surround the 224 line, but having braces looks a bit clearer.

Reply:
@Shan1024 I am talking about `err=e` and `err = e`

Reply:
Ah, didnt see that.